.github/workflows/build-and-deploy.yml

```diff
@@ -164,7 +164,7 @@ jobs:
     concurrency:
       group: staging-repo-${{ needs.ci-preparation.outputs.build-ref }}
       cancel-in-progress: true
-    if: needs.ci-preparation.outputs.may-deploy
+    if: needs.ci-preparation.outputs.may-deploy == 'true'
```

Comment:
restore

Reply:
I restored it, but then staging repo is executed in forks and pull-requests, check this run:
https://github.com/Zimbrando/Alchemist/actions/runs/7871228546

Adding that check prevent this to happen, even though I do not understand why

Reply:
Without ${{ https://github.com/AlchemistSimulator/Alchemist/pull/3120/commits/26ad51802a05370e6d8708c418e5ba6d9fae706b

Reply:
You are right. Github Action's language is *terrible*